projects/storefrontlib/src/cms-components/user/login/login.component.ts

```diff
@@ -1,33 +1,18 @@
 import { Component } from '@angular/core';
-import { AuthService, User, UserService } from '@spartacus/core';
+import { User, UserService } from '@spartacus/core';
 import { Observable } from 'rxjs';
-import { filter, map, switchMap } from 'rxjs/operators';
-import { LoginComponentService } from './login.component.service';
+import { map } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-login',
   templateUrl: './login.component.html',
 })
 export class LoginComponent {
-  constructor(
-    private auth: AuthService,
-    private userService: UserService,
-    private loginService: LoginComponentService
-  ) {}
+  constructor(private userService: UserService) {}
 
   get user$(): Observable<User> {
-    return this.auth.getUserToken().pipe(
-      map(token => {
-        if (token && !!token.access_token && !this.loginService.isLogin) {
-          this.loginService.isLogin = true;
-          this.userService.load(token.userId);
-        } else if (token && !token.access_token && this.loginService.isLogin) {
-          this.loginService.isLogin = false;
-        }
-        return token;
-      }),
-      filter(token => token && !!token.access_token),
-      switchMap(() => this.userService.get())
-    );
+    return this.userService
```

Comment:
So you validate that the user is signed in if there is a user instead of checking for the token? Why exactly?

Reply:
I can think of 2 reasons at the moment.

Since the user details is now part of the authentication data, and guaranteed to be present when a user logs in, its the kind of logic we can rely on.  It is even preferable because the user identifier that is used in the OCC calls comes from the user details (wether it's the uid or the customer id)

Also, it's the kind of logic that will be required when we implement ASM support. The token will belong to the customer support agent, but the user details will be populated with the customer being impersonated.